src/utils/survey.ts

```diff
@@ -0,0 +1,238 @@
+        if (optOutDateString) {
+            // Skip if opted out within the last REARM_AFTER_DAYS days
+            const optOutDate = new Date(optOutDateString);
+            if (!REARM_OPT_OUT || optOutDate.getTime() >= rearmAfterDate.getTime()) {
+                context.telemetry.properties.isCandidate = (isCandidate = false).toString();
+                return;
+            }
+        }
+
+        // Prompt only once per day,
+        // LAST_SESSION_DATE_KEY will be set to the last date the user was prompted
+        const lastSessionDate = new Date(ext.context.globalState.get(LAST_SESSION_DATE_KEY, new Date(0).toISOString()));
+        if (PROMPT_DATE_ONLY_ONCE && today.toDateString() === lastSessionDate.toDateString()) {
+            context.telemetry.properties.isCandidate = (isCandidate = false).toString();
+            return;
+        }
+
+        // Count sessions and decide if the user is a candidate
+        const sessionCount = ext.context.globalState.get(SESSION_COUNT_KEY, 0) + 1;
+        await ext.context.globalState.update(LAST_SESSION_DATE_KEY, today);
+        await ext.context.globalState.update(SESSION_COUNT_KEY, sessionCount);
+        if (sessionCount < MIN_SESSIONS_BEFORE_PROMPT) {
+            context.telemetry.properties.isCandidate = (isCandidate = false).toString();
+            return;
+        }
+
+        // If the user is a candidate (has not opted out or participated for the current version),
+        // decide randomly with given probability
+        isCandidate = ext.context.globalState.get(IS_CANDIDATE_KEY, false) || Math.random() < PROBABILITY;
+        context.telemetry.properties.isCandidate = isCandidate.toString();
+        await ext.context.globalState.update(IS_CANDIDATE_KEY, isCandidate);
+    });
+}
+
+export async function surveyPromptIfCandidate(
+    experience: ExperienceKind = ExperienceKind.NoSQL,
+    triggerAction?: string,
+): Promise<void> {
+    if (wasPromptedInSession) {
+        return;
+    }
+    await callWithTelemetryAndErrorHandling('survey.prompt', async (context: IActionContext) => {
+        const isCandidate = await getIsSurveyCandidate();
+        context.telemetry.properties.isCandidate = isCandidate.toString();
+        context.telemetry.properties.experience = experience;
+        context.telemetry.properties.triggerAction = triggerAction;
+        context.telemetry.properties.userAsked = 'false'; // this will be set to 'true' later if the user interacts with the prompt
+        wasPromptedInSession = true; // disarm for the rest of the session
+        if (!isCandidate) {
+            return;
+        }
+
+        const extensionVersion = (ext.context.extension.packageJSON as { version: string }).version;
+        const date = new Date().toDateString();
+        const surveyUrl = experience === ExperienceKind.Mongo ? NPS_MONGO_SURVEY_URL : NPS_NOSQL_SURVEY_URL;
+
+        const take = {
+            title: localize('azureResourceGroups.takeSurvey', 'Take Survey'),
```

Comment:
please use `vscode.l10n.t()`

Reply:
Since your localisation PR is not merged yet, I'd rather avoid this change here and then update your PR after merging.